scripts/ecleanse.lic

```diff
@@ -0,0 +1,576 @@
+    
+    echo "right: #{right} left: #{left}"
+    
+    look_for = GameObj.right_hand != right ? right : left
+  
+    echo "########## DISARMED!!! ##########"
+    echo "## Room#: #{Room.current.id}"
+    echo "## Room Name: #{Room.current.title}"
+    echo "## Looking for: #{look_for.name}"
+    echo "########## DISARMED!!! ##########"
+    
+    waitrt?
+    line = nil
+    if Spell[218].active?
+      line = dothistimeout "tell servant recover", 5, /flickers for a moment and manifests|has no personal recollection/
+      if line =~ /flickers for a moment and manifests/
+        servant_recover = true 
+      else
+        servant_recover = false
+      end
+      if servent_recover
+        @disarmed = false
+        return
+      end
+    end
+  
+    #Fixme: Others?
+    if Spell[619].known? #mass calm
+			fput "release"
+      Spell[619].cast
+			fput "prep 619"
+			fput "cast"
+		elsif Spell[709].known? #grasp of the grave
+			fput "release"
+      Spell[709].cast		
+		elsif Spell[9811].known? #symbol of sleep
+			fput "sym of sleep"
+		elsif Spell[919].known? #wizard shield
+			fput "release"
+      Spell[919].cast		
+		end
+  
+    fput "stance defensive"
+		line = nil
+		empty_hands
+		until line =~ /You spy (?:an|a) (?:.*) and recover it\!/
+			until(kneeling?)
+				waitrt?
+				fput "kneel"
+				exit if checkdead
+			end
+			waitrt?
+			line = dothistimeout "recover item", 5, /You find nothing recoverable\.|You spy (?:an|a) (?:.*) and recover it\!|...wait (?:.*) seconds\./
+		end
+    
+    @disarmed = false
+		waitrt?
+		fill_hands
+    
+    fput "stand" until(standing?)
+    
+    return true
+    
+  end
+  
+  def self.start_up
+    
+   
+    Ecleanse.set_hooks if CharSettings['recover_disarmed']
+ 
+    dispellable_debuffs = ["Confusion", "Vertigo", "Sounds", "Thought Lash"]
+    grounded_debuffs = ["Rooted", "Pressed"]
+    dispel = [417, 1218, 119].map { |s| Spell[s] }.find { |s| s.known? }
+    webs = [906,417,209,119].map { |s| Spell[s] }.find { |s| s.known? }
+    
+    
+     loop{
+    
+      if checkpoison && CharSettings['cleanse_poison']
+        Ecleanse.pause_scripts
+        Ecleanse.remove_poison
+        Ecleanse.unpause_scripts
+      elsif checkdisease && CharSettings['cleanse_disaese']
+        Ecleanse.pause_scripts
+        Ecleanse.remove_disease
+        Ecleanse.unpause_scripts
+      elsif !Effects::Debuffs.to_h.empty? && (CharSettings['cleanse_magical'] || CharSettings['cleanse_grounded'])
+        dispellable_debuffs.each { |debuff|
+          if Effects::Debuffs.active?(debuff)
+            Ecleanse.pause_scripts
+            dispel.force_incant(arg_options = "channel open") if dispel.affordable?
+            Ecleanse.unpause_scripts
+          end
+        } 
+       
+        grounded_debuffs.each { |debuff|
+          if Effects::Debuffs.active?(debuff)
+            Ecleanse.pause_scripts
+            waitrt?
+            waitcastrt?
+            fput "cman retreat"
```

Comment:
Need some checks to verify result here. Got put into kneeling, and it spammed cman retreat.

`[ecleanse]>cman retreat`
`You might want to stand up first.`
`!s>`
`[ecleanse]>cman retreat`
`You might want to stand up first.`